compiler/rustc_macros/src/diagnostics/utils.rs

```diff
@@ -621,3 +619,69 @@ impl quote::IdentFragment for SubdiagnosticKind {
         None
     }
 }
+
+/// Wrapper around `synstructure::BindStyle` which implements `Ord`.
+#[derive(PartialEq, Eq)]
+pub(super) struct OrderedBindStyle(pub(super) BindStyle);
+
+impl OrderedBindStyle {
+    /// Is `BindStyle::Move` or `BindStyle::MoveMut`?
+    pub(super) fn is_move(&self) -> bool {
+        matches!(self.0, BindStyle::Move | BindStyle::MoveMut)
+    }
+}
+
+impl Ord for OrderedBindStyle {
+    fn cmp(&self, other: &Self) -> Ordering {
+        // If both are moves then they are equal, and if both aren't moves, then they both
+        // must be refs, and are also equal.
+        if self.is_move() && other.is_move() {
```

Comment:
Shouldn't this be `==` instead of `&&`? It might makes sense to use a `(bool, bool)` match here instead for clarity

Reply:
Fixed!